rhino/src/main/java/org/mozilla/javascript/typedarrays/NativeTypedArrayView.java

```diff
@@ -151,6 +150,355 @@ private static int toIndex(double num) {
+                0,
+                (Context lcx, Scriptable lscope, Scriptable thisObj, Object[] args) -> {
+                    NativeTypedArrayView<?> self = realThis.realThis(thisObj);
+                    return new NativeArrayIterator(lscope, self, ARRAY_ITERATOR_TYPE.KEYS);
+                },
+                DONTENUM,
+                DONTENUM | READONLY);
+        constructor.definePrototypeMethod(
+                scope,
+                "lastIndexOf",
+                1,
+                (Context lcx, Scriptable lscope, Scriptable thisObj, Object[] args) ->
+                        js_lastIndexOf(lcx, lscope, thisObj, args, realThis),
+                DONTENUM,
+                DONTENUM | READONLY);
+        constructor.definePrototypeMethod(
+                scope,
+                "map",
+                1,
+                (Context lcx, Scriptable lscope, Scriptable thisObj, Object[] args) -> {
+                    NativeTypedArrayView<?> self = realThis.realThis(thisObj);
+                    Object array =
+                            ArrayLikeAbstractOperations.iterativeMethod(
+                                    lcx, IterativeOperation.MAP, lscope, thisObj, args);
+                    return self.typedArraySpeciesCreate(lcx, lscope, new Object[] {array}, "map");
+                },
+                DONTENUM,
+                DONTENUM | READONLY);
+        constructor.definePrototypeMethod(
+                scope,
+                "reduce",
+                1,
+                (Context lcx, Scriptable lscope, Scriptable thisObj, Object[] args) -> {
+                    NativeTypedArrayView<?> self = realThis.realThis(thisObj);
+                    return ArrayLikeAbstractOperations.reduceMethod(
+                            lcx, ReduceOperation.REDUCE, lscope, self, args);
+                },
+                DONTENUM,
+                DONTENUM | READONLY);
+        constructor.definePrototypeMethod(
+                scope,
+                "reduceRight",
+                1,
+                (Context lcx, Scriptable lscope, Scriptable thisObj, Object[] args) -> {
+                    NativeTypedArrayView<?> self = realThis.realThis(thisObj);
+                    return ArrayLikeAbstractOperations.reduceMethod(
+                            lcx, ReduceOperation.REDUCE_RIGHT, lscope, self, args);
+                },
+                DONTENUM,
+                DONTENUM | READONLY);
+        constructor.definePrototypeMethod(
+                scope,
+                "reverse",
+                0,
+                (Context lcx, Scriptable lscope, Scriptable thisObj, Object[] args) ->
+                        js_reverse(lcx, lscope, thisObj, args, realThis),
+                DONTENUM,
+                DONTENUM | READONLY);
+        constructor.definePrototypeMethod(
+                scope,
+                "set",
+                0,
+                (Context lcx, Scriptable lscope, Scriptable thisObj, Object[] args) -> {
```

Comment:
Minor, but this seems like a bit too much code to be embedded inline like this, and it'd probably benefit from being a separate method.